grammar.js

```diff
@@ -700,8 +715,17 @@ module.exports = grammar({
       ),
     )),
 
-    module_destructuring_expression: $ => seq(
-      'module', '(', choice($.module_identifier, $.module_identifier_path), ')'
+    module_pack_expression: $ => seq(
+      'module', '(',
+      choice(
+        seq(
+          field('name', choice($.module_identifier, $.module_identifier_path)),
+          optional(field('signature', seq(':', $.module_identifier))),
+          optional(field('definition', seq('(', $.module_unpack, ')')))
+        ),
+        seq(field('definition', $.block), ':', field('signature', $.module_identifier))
+      ),
+      ')'
```

Comment:
This one looks bulky. I suggest splitting it by extracting two root seq choices into two separate rules and referring them here.

I also pretty sure that for a “signature” one can use not just `$.module_identifier` but also `$.module_identifier_path` or even a `$.module_expression`.

And I can’t imagine an example when the “definition” is used. I’ve tried to comment this line out and saw no tests failing.

Reply:
Done 812b1ab42ba34e1cd85ad2d47224bbd39b572b9b